### ISSUE TITLE
Introduce ginkgo labels

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/cluster-api#9620]
Since ginkgo v2 tests can be selected via labels. I think we should take a closer look at that feature and then use it in Cluster API instead of ginkgo focus / skip.

For more details, see: https://groups.google.com/a/kubernetes.io/g/dev/c/DTFEng143NY

**Comment by fabriziopandini** [comment]
+1, especially if we follow the same approach suggested for k/k that will ensure a transition period for everyone using the CAPI E2E test framework

**Comment by sbueringer** [comment]
Do we need code from k/k to implement ginkgo labels?

**Comment by cahillsf** [comment]
not necessarily -- that was the alternative approach i was thinking is "just" using the Ginkgo functionality.  

seems they have implemented some nice abstractions around it in k/k though (this valid features thing for example: https://github.com/pohly/kubernetes/blob/338fe7ad55cad858284d4c96df86d3cee9c7d29c/test/e2e/feature/feature.go#L136 or this `WithLabel` wrapper: https://github.com/pohly/kubernetes/blob/338fe7ad55cad858284d4c96df86d3cee9c7d29c/test/e2e/framework/ginkgowrapper.go#L513-L518)

**Comment by sbueringer** [comment]
Yeah. I'm just wondering if we need this sort of framework considering we only have very few labels :)

**Comment by cahillsf** [comment]
@sbueringer 👋 took a flip through the jobs running on main and it seems like they have been running as expected since the cutover

there is one small followup to remove release branches in the template `if` statements as we cut new release branches and older ones fall out of support (and eventually remove the if/else to just use the updated envvar).  

this could be captured as a generalized instruction in the [release CI team docs](https://github.com/kubernetes-sigs/cluster-api/tree/main/docs/release/role-handbooks/ci-signal#setup-jobs-and-dashboards-for-a-new-release-branch), i.e.:
> update any conditional branches in template logic for releases that are no longer supported

ill open a PR to add this and then close out this issue? wdyt?

**Comment by sbueringer** [comment]
Sounds good

**Comment by sbueringer** [comment]
> took a flip through the jobs running on main and it seems like they have been running as expected since the cutover

Did the same, agree - all good

**Comment by chrischdi** [comment]
PR got merged :-)

/close

Thanks @cahillsf ! 🎉

**Comment by sbueringer** [comment]
Thank you very much @cahillsf !!

and @chrischdi for helping with ci-e2e-lib.sh!
